R/verb-group_by.R

```diff
@@ -37,20 +37,12 @@ group_by.tbl_lazy <- function(.data, ..., .add = FALSE, add = NULL, .drop = TRUE
     stop("`.drop` is not supported with database backends", call. = FALSE)
   }
 
-  if (length(dots) == 0) {
-    if (.add) {
-      return(.data)
-    } else {
-      return(dplyr::ungroup(.data))
-    }
-  }
+  groups <- dplyr::group_by_prepare(.data, !!!dots, .add = .add)
+  names <- purrr::map_chr(groups$groups, as_string)
 
-  if (".add" %in% names(formals("group_by"))) {
-    groups <- dplyr::group_by_prepare(.data, !!!dots, .add = .add)
-  } else {
-    groups <- dplyr::group_by_prepare(.data, !!!dots, add = .add)
+  if (can_group_by_can_return_early(.data, groups$data, groups$groups)) {
```

Comment:
Should we inline this function? I'm not sure how to read `setequal(.data, ...)` -- should we do `setequal(group_vars(.data), ...)` or the like?

Reply:
I realised that this can be done much simpler anyway 😄